quickwit-config/src/index_config.rs

```diff
@@ -19,70 +19,203 @@
+    }
+}
+
+impl Default for IndexingResources {
+    fn default() -> Self {
+        Self {
+            num_threads: Self::default_num_threads(),
+            heap_size: Self::default_heap_size(),
+        }
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
 pub struct MergePolicy {
+    #[serde(default = "MergePolicy::default_demux_factor")]
     pub demux_factor: usize,
+    #[serde(default = "MergePolicy::default_merge_factor")]
     pub merge_factor: usize,
+    #[serde(default = "MergePolicy::default_max_merge_factor")]
     pub max_merge_factor: usize,
-    pub min_level_num_docs: usize,
 }
 
-#[derive(Debug, Serialize, Deserialize, PartialEq)]
+impl MergePolicy {
+    fn default_demux_factor() -> usize {
+        8
+    }
+
+    fn default_merge_factor() -> usize {
+        10
+    }
+
+    fn default_max_merge_factor() -> usize {
+        12
+    }
+}
+
+impl Default for MergePolicy {
+    fn default() -> Self {
+        Self {
+            demux_factor: Self::default_demux_factor(),
+            merge_factor: Self::default_merge_factor(),
+            max_merge_factor: Self::default_max_merge_factor(),
+        }
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
 pub struct IndexingSettings {
-    pub demux_field: String,
-    pub timestamp_field: String,
+    #[serde(default)]
+    pub demux_enabled: bool,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub demux_field: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub timestamp_field: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub sort_field: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub sort_order: Option<SortOrder>,
+    #[serde(default = "IndexingSettings::default_commit_timeout_secs")]
     pub commit_timeout_secs: usize,
     /// The maximum number of documents allowed in a split.
+    #[serde(default = "IndexingSettings::default_split_max_num_docs")]
     pub split_max_num_docs: usize,
+    #[serde(default = "IndexingSettings::default_merge_enabled")]
+    pub merge_enabled: bool,
+    #[serde(default)]
     pub merge_policy: MergePolicy,
+    #[serde(default)]
     pub resources: IndexingResources,
 }
 
-#[derive(Debug, Serialize, Deserialize, PartialEq)]
+impl IndexingSettings {
+    pub fn commit_timeout(&self) -> Duration {
+        Duration::from_secs(self.commit_timeout_secs as u64)
+    }
+
+    fn default_commit_timeout_secs() -> usize {
+        60
+    }
+
+    fn default_split_max_num_docs() -> usize {
+        10_000_000
+    }
+
+    fn default_merge_enabled() -> bool {
+        false
+    }
+
+    pub fn sort_by(&self) -> SortBy {
+        if let Some(field_name) = self.sort_field.clone() {
+            let order = self.sort_order.clone().unwrap_or_else(SortOrder::default);
+            return SortBy::FastField { field_name, order };
+        }
+        SortBy::DocId
+    }
+
+    pub fn for_test() -> Self {
```

Comment:
I'll have to hide those in a follow-up PR.

Reply:
```suggestion
    // TODO(guilload) Hide this method if possible.
    pub fn for_test() -> Self {
```